src/NuGet.Core/NuGet.Commands/RestoreCommand/RequestFactory/DependencyGraphSpecRequestProvider.cs

```diff
@@ -86,6 +87,7 @@ private IReadOnlyList<RestoreSummaryRequest> GetRequestsFromItems(RestoreArgs re
                 // Limiting to processor count reduces task context switching which is better
                 MaxDegreeOfParallelism = Environment.ProcessorCount
             };
+            var dictionary = new Dictionary<string, string>(PathUtility.GetStringComparerBasedOnOS());
```

Comment:
I think this variable's name can be improved. I can see from its type that it's a dictionary, but currently the variable name doesn't give a hint what it's a dictionary of, or how it's used.